core/src/main/resources/lib/form/select/select.js

```diff
@@ -30,13 +35,24 @@ function updateListBox(listBox,url,config) {
 
         if (originalOnSuccess!=undefined)
             originalOnSuccess(rsp);
-    },
-    config.onFailure = function(rsp) {
-        // deleting values can result in the data loss, so let's not do that
-//        var l = $(listBox);
-//        l.options[0] = null;
-    }
+    };
+    config.onFailure = function (rsp) {
+        l.removeClassName("select-ajax-pending");
+        status.innerHTML = rsp.responseText;
+        if (status.firstChild) {
+            status.firstChild.setAttribute('data-select-ajax-error', 'true')
+        }
+        Behaviour.applySubtree(status);
+        // deleting values can result in the data loss, so let's not do that unless instructed
+        var header = rsp.getResponseHeader('X-Jenkins-Select-Error');
```

Comment:
I'm fine with that. Once we start following HTTP2 rules, we will need to either make Stapler to do it for us OR to perform a bulk update in the core and plugins. I prefer the first option